std-bits/table/src/main/java/org/enso/table/excel/xssfreader/XSSFReaderRow.java

```diff
@@ -0,0 +1,126 @@
+    this.data = data;
+    this.use1904Dates = use1904Dates;
+  }
+
+  @Override
+  public int getFirstColumn() {
+    return data.firstKey();
+  }
+
+  @Override
+  public int getLastColumn() {
+    return data.lastKey();
+  }
+
+  @Override
+  public Cell get(int column) {
+    // Not supported as we don't have the underlying Apache POI Cell object.
+    return null;
+  }
+
+  @Override
+  public Object getCellValue(int column) {
+    var cell = data.get((short) column);
+    if (cell == null) {
+      return null;
+    }
+
+    var dataType = cell.dataType();
+    return switch (dataType) {
+      case BOOL -> cell.getBooleanValue();
+      case DATE -> LocalDateTime.parse(cell.strValue()); // Don't believe used by Excel.
+      case INLINE_STRING, SST_STRING, FORMULA_STRING -> cell.strValue();
+      case INTEGER -> cell.getIntegerValue();
+      case NUMBER -> {
+        double dbl = cell.getNumberValue();
+        long longVal = (long) dbl;
+        if (dbl == longVal) {
+          yield (long) dbl;
+        } else {
+          yield dbl;
+        }
+      }
+      case OLE_DATE -> cell.getDateValue(use1904Dates);
+      case OLE_DATETIME -> cell.getDateTimeValue(use1904Dates);
+      case ERROR -> null;
+    };
+  }
+
+  @Override
+  public String getCellText(int column) {
+    var cell = data.get((short) column);
+    if (cell == null) {
+      return "";
+    }
+
+    var dataType = cell.dataType();
+    return switch (dataType) {
+      case NUMBER, OLE_DATETIME, OLE_DATE, INTEGER -> {
+        // Special handling for Number or Date cells as want to keep formatting.
+        var formatText = cell.format();
+        if (formatText == null || formatText.isEmpty()) {
+          yield cell.strValue();
+        }
+        yield formatter.formatRawCellContents(cell.getNumberValue(), -1, formatText, use1904Dates);
+      }
+      case BOOL -> cell.getBooleanValue() ? "TRUE" : "FALSE";
+      default -> cell.strValue();
+    };
+  }
+
+  @Override
+  public boolean isEmpty(int column) {
+    var cell = data.get((short) column);
+    return cell == null || cell.strValue().isEmpty();
+  }
+
+  @Override
+  public boolean isEmpty(int start, int end) {
+    int currentEnd = end == -1 ? getLastColumn() : end;
+    for (int column = Math.max(getFirstColumn(), start);
+        column <= Math.min(getLastColumn(), currentEnd);
+        column++) {
+      if (!isEmpty(column)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public String[] getCellsAsText(int startCol, int endCol) {
+    int currentEndCol = endCol == -1 ? getLastColumn() : endCol;
+
+    String[] output = new String[currentEndCol - startCol + 1];
+    for (int col = startCol; col <= currentEndCol; col++) {
+
+      var cell = data.get((short) col);
+      if (cell != null) {
+        var dataType = cell.dataType();
+        if (dataType != XSSFReaderSheetXMLHandler.XSSDataType.INLINE_STRING
```

Comment:
What is the purpose of this check? Why do we return null if it fails? Can we get some doc explaining it?